test/unit/dfc_test.py

```diff
@@ -117,59 +123,61 @@ def test_union_of_two(self):
         dfc2.store("second", second)
 
         final_dfc = dfc1.union(dfc2)
-        self.assertEqual(["first"], dfc1.get_table_names())
-        self.assertEqual(["second"], dfc2.get_table_names())
-        self.assertEqual(["first", "second"], final_dfc.get_table_names())
+        assert dfc1.get_table_names() == ["first"]
+        assert dfc2.get_table_names() == ["second"]
+        assert final_dfc.get_table_names() == ["first", "second"]
+
         pd.testing.assert_frame_equal(first, final_dfc.get("first"))
         pd.testing.assert_frame_equal(second, final_dfc.get("second"))
 
-    def test_overlapping_union_disallowed(self):
+    def test_overlapping_union_disallowed(self, temp_dir):
         """Tests that merging two dataframes with the same table is not permitted."""
-        dfc1 = DataFrameCollection(storage_path=self.temp_dir)
-        dfc2 = DataFrameCollection(storage_path=self.temp_dir)
+        dfc1 = DataFrameCollection(storage_path=temp_dir)
+        dfc2 = DataFrameCollection(storage_path=temp_dir)
 
         dfc1.store("x", pd.DataFrame())
         dfc1.store("y", pd.DataFrame())
         dfc2.store("y", pd.DataFrame())
         dfc2.store("z", pd.DataFrame())
-        self.assertRaises(dfc.TableExistsError, dfc1.union, dfc2)
 
-    def test_to_dict(self):
+        with pytest.raises(dfc.TableExistsError):
+            dfc1.union(dfc1)
```

Comment:
Is this supposed to be unioning `dfc1` with itself? Or with `dfc2`?

Reply:
This is a bug. The original code unioned `dfc1` and `dfc2`.